### PR TITLE
e2e.sh: Update to kind v0.10.0

### PR DESCRIPTION
**Pull request by mrueg**
**What this PR does / why we need it**:
Update kind to v0.10.0.

### CODE REVIEW

File: tests/e2e.sh [hunk 1]
```diff
@@ -23,13 +23,14 @@ case $(uname -m) in
 	*)		ARCH="$(uname -m)";;
 esac
 
-KUBERNETES_VERSION=v1.19.1
+KUBERNETES_VERSION="v1.20.2"
```
Comment (tariq1890) [inline tests/e2e.sh:26]: I actually prefer bundling all the k8s updates together in one PR.

So, can we update the go deps of k8s and the compatibility matrix ?

**Comment by tariq1890** [comment]
Thanks for the PR @mrueg

File: tests/e2e.sh (hunk 1 above)
Comment (mrueg) [reply tests/e2e.sh:26]: Sure thing

Comment (mrueg) [reply tests/e2e.sh]: Thanks for letting me know! I haven't started, so I'll limit this one to kind v0.10.0 only.

**Review by lilic — COMMENTED** [review]
Thank you!! 🙇‍♀️ 

/lgtm

/hold

@tariq1890 any objections, feel free to remove the hold :)

File: tests/e2e.sh (hunk 1 above)
Comment (tariq1890) [reply tests/e2e.sh]: Whoops. Missed that. Thanks @lilic

**Pull request merged**